### ISSUE TITLE
Duplicated and invalid local repositories after building with vsxmake when using anti-slashes

### ISSUE THREAD

**Issue by WubiCookie** [xmake-io/xmake#2587]
### Xmake Version

v2.6.9+HEAD.22ecc0156 (dev)

### Operating System Version and Architecture

Windows 10 version 1909

### Describe Bug

The local repos are messed up after I try to build with vsxmake2022 and the configuration is lost.

### Expected Behavior

The local repos should be left untouched as well as the configuration.

### Project Configuration

I have two local repos:
```lua
add_repositories("local-repo xmake-packages")
add_repositories("iconeus-repo https://github.com/Iconeus/xmake-repo.git")
```
The first one is cloned from SVN and contains pre-build binaries, and the second is a private package repository.

### Additional Information and Error Logs

TL;DR: vsxmake does not seem to like antislashes at all in options

___________________________

This is the initial state, the repos are correct, the project is configured, it builds and runs fine, the repos are untouched before and after generating the vs solution.

```console
> xmake repo -l
local repositories:
    local-repo xmake-packages
    iconeus-repo https://github.com/Iconeus/xmake-repo.git

global repositories:
    build-artifacts https://gitlab.com/xmake-mirror/build-artifacts.git main
    xmake-repo https://gitlab.com/tboox/xmake-repo.git master
    builtin-repo C:\Users\Charles\xmake\repository

5 repositories were found!

> xmake
checking for Microsoft Visual Studio (x64) version ... 2022
checking for Microsoft C/C++ Compiler (x64) version ... 19.31.31104
updating repositories .. ok
checking for Qt SDK directory ... C:/Users/Charles/Documents/work/qt6-build-msvc/qtbase
checking for Qt SDK version ... 6.3.0
...
[100%]: build ok!

> xmake repo -l
local repositories:
    local-repo xmake-packages
    iconeus-repo https://github.com/Iconeus/xmake-repo.git

global repositories:
    build-artifacts https://gitlab.com/xmake-mirror/build-artifacts.git main
    xmake-repo https://gitlab.com/tboox/xmake-repo.git master
    builtin-repo C:\Users\Charles\xmake\repository

5 repositories were found!

> xmake project -yk vsxmake
...
create ok!

> xmake repo -l
local repositories:
    local-repo xmake-packages
    iconeus-repo https://github.com/Iconeus/xmake-repo.git

global repositories:
    build-artifacts https://gitlab.com/xmake-mirror/build-artifacts.git main
    xmake-repo https://gitlab.com/tboox/xmake-repo.git master
    builtin-repo C:\Users\Charles\xmake\repository

5 repositories were found!
```

I reload the solution and build it, I get an error

```
error : Designtime build failed for project 'C:\Users\Charles\Documents\work\IcoStudio\vsxmake2022\IcoStudioInterface\IcoStudioInterface.vcxproj' configuration 'debug|x64'. IntelliSense might be unavailable.
  Set environment variable TRACEDESIGNTIME = true and restart Visual Studio to investigate.
```

I have to cancel the compilation and build again. It launches, but a configuration is missing.
`xmake config -y  -P . -p windows -m debug -a x64 -o "build"`
should be
`xmake config -y  -P . -p windows -m debug -a x64 -o "build" "--matlabroot=D:\\Program Files\\MATLAB\\MATLAB Runtime\\v911"`

```
Build started...
1>------ Build started: Project: IcoStudioInterface, Configuration: debug x64 ------
1>$xmake config -y  -P . -p windows -m debug -a x64 -o "build"
1>checking for Microsoft Visual Studio (x64) version ... 2022
1>checking for Microsoft C/C++ Compiler (x64) version ... 19.31.31104
1>checking for Qt SDK directory ... C:/Users/Charles/Documents/work/qt6-build-msvc/qtbase
1>checking for Qt SDK version ... 6.3.0
1>generating src\IcoStudioInterface\Config.hpp.in ... ok
1>$xmake build -y  -P . -w "IcoStudioInterface"
1>[ 85%]: compiling.debug build\.gens\IcoStudioSoftware\windows\unity_build\unity_1.cpp
1>[ 87%]: archiving.debug IcoStudioSoftware.lib
1>[ 89%]: compiling.debug build\.gens\IcoStudioInterface\windows\unity_build\unity_1.cpp
1>[ 90%]: compiling.debug build\.gens\IcoStudioRendering\windows\unity_build\unity_1.cpp
1>[ 95%]: archiving.debug IcoStudioRendering.lib
1>[ 97%]: linking.debug IcoStudio.exe
1>[100%]: build ok!
========== Build: 1 succeeded, 0 failed, 0 up-to-date, 0 skipped ==========
```

It builds fine but the runtime can't find the dlls provided with the `matlabroot` option (although it also provides include dirs, link dir and links?).
I reconfigure and rebuild the project.

```console
> xmake config -y --matlabroot="D:\Program Files\MATLAB\MATLAB Runtime\v911"

> xmake project -yk vsxmake
...
create ok!
```

Then I reload the solution and build. The option is here, but the repos are messed up.

```
Build started...
1>------ Build started: Project: IcoStudioInterface, Configuration: debug x64 ------
1>$xmake config -y  -P . -p windows -m debug -a x64 -o "build" "--matlabroot=D:\\Program Files\\MATLAB\\MATLAB Runtime\\v911"
1>checking for Microsoft Visual Studio (x64) version ... 2022
1>checking for Microsoft C/C++ Compiler (x64) version ... 19.31.31104
1>  => install libiconeus 0.1.3 .. failed
1>
1>EXEC : error : Cloning into 'C:\Users\Charles\Documents\work\IcoStudio\.xmake\windows\x64\repositories\iconeus-repo'...
1>fatal: Remote branch master not found in upstream origin
1>if you want to get more verbose errors, please see:
1>  -> C:\Users\Charles\AppData\Local\.xmake\cache\packages\2207\l\libiconeus\0.1.3\installdir.failed\logs\install.txt
1>EXEC : error : install failed!
1>C:\Users\Charles\xmake\plugins\project\vsxmake\vsproj\Xmake.targets(97,5): error MSB3073: The command "..." exited with code -1.
1>Done building project "IcoStudioInterface.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

```console
> xmake repo -l
local repositories:
    local-repo C:\Users\Charles\Documents\work\IcoStudio\xmake-packages
    iconeus-repo C:\Users\Charles\Documents\work\IcoStudio\.xmake\windows\x64\repositories\iconeus-repo
    local-repo C:\Users\Charles\Documents\work\IcoStudio\xmake-packages
    iconeus-repo C:\Users\Charles\Documents\work\IcoStudio\.xmake\windows\x64\repositories\iconeus-repo

global repositories:
    build-artifacts https://gitlab.com/xmake-mirror/build-artifacts.git main
    xmake-repo https://gitlab.com/tboox/xmake-repo.git master
    builtin-repo C:\Users\Charles\xmake\repository

7 repositories were found!
```

I remove the repos, it compiles and runs well, we're back to the initial state.

```console
> xmake repo -r local-repo
remove local repository(local-repo): ok!
> xmake repo -r iconeus-repo
remove local repository(iconeus-repo): ok!
> xmake repo -l
local repositories:
    local-repo xmake-packages
    iconeus-repo https://github.com/Iconeus/xmake-repo.git

global repositories:
    build-artifacts https://gitlab.com/xmake-mirror/build-artifacts.git main
    xmake-repo https://gitlab.com/tboox/xmake-repo.git master
    builtin-repo C:\Users\Charles\xmake\repository

5 repositories were found!

> xmake
...
[100%]: build ok!

> xmake r
>
```

___________________________

Now I reconfigure with forward-slashes in the option

```console
> xmake config -y --matlabroot="D:/Program Files/MATLAB/MATLAB Runtime/v911"

> xmake project -yk vsxmake
...
create ok!
```

And everything works fine both from the command line and from vsxmake without any error.

**Comment by waruqi** [comment]
> xmake config -y --matlabroot="D:\Program Files\MATLAB\MATLAB Runtime\v911"

you should use `\\` instead of `\`

**Comment by WubiCookie** [comment]
No, the behavior should be consistent between xmake CLI and vsxmake

**Comment by waruqi** [comment]
> > xmake config -y --matlabroot="D:\Program Files\MATLAB\MATLAB Runtime\v9

The cli terminal shell will escapes `\`, so you have to use `\\` instead of `\`, which has nothing to do with xmake. 

xmake already gets the wrong parameter value from the terminal, before it has done anything. 
xmake has no chance to change anything, unless the user corrects the input parameters.

**Comment by WubiCookie** [comment]
If xmake gets a wrong parameter value, why does it work in CLI, then?

**Comment by waruqi** [comment]
> 
> It builds fine but the runtime can't find the dlls provided with the matlabroot option (although it also provides include dirs, link dir and links?).
> I reconfigure and rebuild the project.
> xmake config -y --matlabroot="D:\Program Files\MATLAB\MATLAB Runtime\v911"

Didn't you set it in cli?

**Comment by WubiCookie** [comment]
Yes, I set it for the command line in powershell

**Comment by waruqi** [comment]
> Yes, I set it from the command line in powershell

so, you should use

```
xmake config -y --matlabroot="D:\\Program Files\\MATLAB\\MATLAB Runtime\\v911"
```

powetshell will escape `\\`, then xmake will get a valid `\` character.

if you use `\MATLAB`, powershell will escape `\M` amd xmake will get a invalid character or missing it.

This is also the case in the cmd.

**Comment by WubiCookie** [comment]
But it doesn't, tho

![image](https://user-images.githubusercontent.com/17192695/180446182-b61e8f6e-35a1-44ef-b1c4-f86890201507.png)

![image](https://user-images.githubusercontent.com/17192695/180446022-e1160ba5-3842-475f-9856-3447c747a3cd.png)

It compiles and runs fine

**Comment by waruqi** [comment]
you can edit `Xmake.Custom.props` file and enable

```
    <!-- Set -v for all xmake call -->
     <XmakeVerbose Condition="'$(XmakeVerbose)' == ''">true</XmakeVerbose>

    <!-- Set -D for all xmake call -->
    <XmakeDiagnosis Condition="'$(XmakeDiagnosis)' == ''">true</XmakeDiagnosis>
```

then give me full logs for installing packages. 

> 1>$xmake config -y  -P . -p windows -m debug -a x64 -o "build" "--matlabroot=D:\\Program Files\\MATLAB\\MATLAB Runtime\\v911"
1>checking for Microsoft Visual Studio (x64) version ... 2022
1>checking for Microsoft C/C++ Compiler (x64) version ... 19.31.31104
1>  => install libiconeus 0.1.3 .. failed
1>
